### ISSUE TITLE
Crash on latest version

### ISSUE THREAD

**Issue by tynansdtm** [KieronQuinn/TapTap#615]
No success since latest update. Crashes immediately when triggered. Restarted, no change.  Uninstalled and downloaded again, no change.
[tap_tap_crash_report_2024-10-26T22_09_08.768.txt](https://github.com/user-attachments/files/17576369/tap_tap_crash_report_2024-10-26T22_09_08.768.txt)
My only action is the tasker event.

**Comment by erikLundstedt** [comment]
I think I have the same issue


Model is: Nothing phone (2)
I also only got a Tasker trigger( and from what I could tell, the Tasker integration is part of the problem)

**Comment by rodrigoswz** [comment]
[tap_tap_crash_report_2024-12-11T21_05_49.704.txt](https://github.com/user-attachments/files/18103802/tap_tap_crash_report_2024-12-11T21_05_49.704.txt)



Same here on my Nothing Phone (2) with Android 15

**Comment by MNLierman** [comment]
Yup, it's constantly crashing. Makes it literally not usable for me because I have to keep relaunching it and restarting the service by toggling it off and on again. I've enabled all permissions, and it happens with all action types. Such a shame because it seemed like a really awesome gesture. Would love to keep using the app. Maybe there is a working fork or someone made a different version.

**Comment by rodrigoswz** [comment]
This crash is happening just with the Tasker Event action. If you use the Tasker Task action instead, it'll work fine. 

Just make sure to disable the Profile (on Tasker app) with the Tap Tap trigger.

**Comment by MNLierman** [comment]
I may not be experiencing the same issue as you then. I'm using Android 15 and it requires the Service Type to be set. Tap Tap is missing this so the service immediately crashes when it is moved to background processes. Meaning it doesn't work for very long after leaving the app. Tap Tap works for maybe 8 min max for me and then it crashes. Requiring me to toggle it off and back on. At worst, it crashes immediately after being triggered.

**Comment by erikLundstedt** [comment]
> Yup, it's constantly crashing. Makes it literally not usable for me because I have to keep relaunching it and restarting the service by toggling it off and on again. I've enabled all permissions, and it happens with all action types. Such a shame because it seemed like a really awesome gesture. Would love to keep using the app. Maybe there is a working fork or someone made a different version. 

Try installing the previous version, 1.6.1 works for me while 1.6.2 has the tasker crash

**Comment by MNLierman** [comment]
> > Yup, it's constantly crashing. Makes it literally not usable for me because I have to keep relaunching it and restarting the service by toggling it off and on again. I've enabled all permissions, and it happens with all action types. Such a shame because it seemed like a really awesome gesture. Would love to keep using the app. Maybe there is a working fork or someone made a different version.
> 
> Try installing the previous version, 1.6.1 works for me while 1.6.2 has the tasker crash

Unfortunately, this doesn't work either. The problem is due to new requirements in Android 15/16 for foreground services to specify explicitly their type on all calls, whenever the service is started. This app doesn't do this, and when I pasted several logs showing it doesn't, the Dev quit replying. He said it does specify the service type, when I showed him it doesn't and that it was crashing due to a missing type in the call, he stopped replying completely – not sure why. Maybe he doesn't believe me, but give it time and I'm sure he'll notice that it doesn't. 

It does make me wonder all of these developers that claim their apps work just fine and Android 15/16, do they not have phones that are getting that update? Samsung is the last one to the party. So I guess if all these developers are using Samsung then that would make sense. 

If I was a Dev I would want to download the latest beta and test my apps. Don't get me wrong, I appreciate this app for what it is, but it is flawed and broken in the latest Android 15 build for Samsung phones. Perhaps the service type call works on other Android 15 devices but it crashes everywhere all the time for Samsung phones on Android 15.

**Comment by TenSeventy7** [comment]
> > > Yup, it's constantly crashing. Makes it literally not usable for me because I have to keep relaunching it and restarting the service by toggling it off and on again. I've enabled all permissions, and it happens with all action types. Such a shame because it seemed like a really awesome gesture. Would love to keep using the app. Maybe there is a working fork or someone made a different version.
> > 
> > 
> > Try installing the previous version, 1.6.1 works for me while 1.6.2 has the tasker crash
> 
> Unfortunately, this doesn't work either. The problem is due to new requirements in Android 15/16 for foreground services to specify explicitly their type on all calls, whenever the service is started. This app doesn't do this, and when I pasted several logs showing it doesn't, the Dev quit replying. He said it does specify the service type, when I showed him it doesn't and that it was crashing due to a missing type in the call, he stopped replying completely – not sure why. Maybe he doesn't believe me, but give it time and I'm sure he'll notice that it doesn't.
> 
> It does make me wonder all of these developers that claim their apps work just fine and Android 15/16, do they not have phones that are getting that update? Samsung is the last one to the party. So I guess if all these developers are using Samsung then that would make sense.
> 
> If I was a Dev I would want to download the latest beta and test my apps. Don't get me wrong, I appreciate this app for what it is, but it is flawed and broken in the latest Android 15 build for Samsung phones. Perhaps the service type call works on other Android 15 devices but it crashes everywhere all the time for Samsung phones on Android 15.

It has a service type set.

https://github.com/KieronQuinn/TapTap/blob/e4dce333fa8f2fe4e73bec64b52fec6dbe34ab7c/app/src/main/AndroidManifest.xml#L28

https://github.com/KieronQuinn/TapTap/blob/e4dce333fa8f2fe4e73bec64b52fec6dbe34ab7c/app/src/main/AndroidManifest.xml#L80

**Comment by SoSeDiK** [comment]
Also getting the crash, seems to be related to https://github.com/joaomgcd/TaskerPluginSample/issues/24
(the reported error *might* be resolved by bumping the taskerpluginlibrary?)

**Comment by EUF15** [comment]
Any update?
